xls/examples/ram.x

```diff
@@ -79,8 +79,14 @@ fn expand_mask<DATA_WIDTH:u32, NUM_PARTITIONS:u32,
  EXPANSION_FACTOR:u32={std::ceil_div(DATA_WIDTH, NUM_PARTITIONS)}>(
   partition_mask:uN[NUM_PARTITIONS]) -> uN[DATA_WIDTH] {
   for (idx, data_mask): (u32, uN[DATA_WIDTH]) in range(u32:0, NUM_PARTITIONS) {
-    let data_mask_segment =
-      flatten(u1[EXPANSION_FACTOR]: [partition_mask[idx +: u1], ...]);
+    let mask_bit: u1 = partition_mask[idx +: u1];
+    let data_mask_segment: uN[EXPANSION_FACTOR] =
+      flatten(u1[EXPANSION_FACTOR]:[mask_bit, ...]);
+
+    let to_flatten: u1[EXPANSION_FACTOR] = u1[EXPANSION_FACTOR]:[partition_mask[idx +: u1], ...];
+    let data_mask_segment': uN[EXPANSION_FACTOR] = flatten(to_flatten);
```

Comment:
What about this change interacted with this? I'm not seeing how the enum stuff applies here.

Reply:
This was exhibiting a bug after I did the constexpr fix. I'll hoist the additional commentary from https://github.com/google/xls/pull/1545#discussion_r1729307861 into the PR description.

The type based on elements was a u1[1] but that was ignoring the ellipsis when EXPANSION_FACTOR was 2.

I made these two variations just to check they produced the same answer & typechecked. It's a little grosser but I figured it wouldn't hurt anything since optimizer will know they're the same -- but I can revert this hunk, now that the bug is fixed w/regression test it shouldn't come back.

Reply:
Ah, makes sense. Sorry, should have looked at the other comment for more context.

I think the example Henner added serves a similar purpose, right? Seems simpler to revert this hunk.

Reply:
PR description is updated and this hunk is reverted.